sdkutil/broadcast.go

```diff
@@ -0,0 +1,155 @@
+package sdkutil
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/input"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authclient "github.com/cosmos/cosmos-sdk/x/auth/client"
+	"github.com/spf13/pflag"
+)
+
+const (
+	broadcastBlockRetryTimeout = 45 * time.Second
+	broadcastBlockRetryPeriod  = time.Second
+)
+
+func BroadcastTX(ctx client.Context, flags *pflag.FlagSet, msgs ...sdk.Msg) error {
+
+	// rewrite of https://github.com/cosmos/cosmos-sdk/blob/ca98fda6eae597b1e7d468f96d030b6d905748d7/client/tx/tx.go#L29
+	// with a different method for broadcast-mode=block:
+	// 1. use BroadcastSync
+	// 2. query for the tx until timeout or the tx is found in a block.
+
+	txf := tx.NewFactoryCLI(ctx, flags)
+
+	if ctx.GenerateOnly {
+		return tx.GenerateTx(ctx, txf, msgs...)
+	}
+
+	txf, err := tx.PrepareFactory(ctx, txf)
+	if err != nil {
+		return err
+	}
+
+	txf, err = adjustGas(ctx, txf, msgs...)
+	if err != nil {
+		return err
+	}
+	if ctx.Simulate {
+		return nil
+	}
+
+	txb, err := tx.BuildUnsignedTx(txf, msgs...)
+	if err != nil {
+		return err
+	}
+
+	ok, err := confirmTx(ctx, txb)
+	if !ok || err != nil {
+		return err
+	}
+
+	err = tx.Sign(txf, ctx.GetFromName(), txb, true)
+	if err != nil {
+		return err
+	}
+
+	txBytes, err := ctx.TxConfig.TxEncoder()(txb.GetTx())
+	if err != nil {
+		return err
+	}
+
+	res, err := doBroadcast(ctx, broadcastBlockRetryTimeout, txBytes)
```

Comment:
Is it possible to pickup `broadcastBlockRetryTimeout` from a flag and have it set on the context? That way if it needs to be increased end users have some control over it?

Reply:
I just changed how `block` is implemented because the previous implementation wasn't working.  I'd rather not add a fourth option and confuse people.